From b989fe1c3e54184b4d12f490a6d56c4939f67928 Mon Sep 17 00:00:00 2001
From: Arun Bharadwaj <arun@gumstix.com>
Date: Mon, 23 May 2016 10:05:28 -0700
Subject: [PATCH 48/57] am335x-pepper-43r: Device tree changes for v6

This contains the following changes:
  * Audio codec has moved to i2c1
  * Different IMU module has been used
  * Use LED triggers for heartbeat and mmc0

Signed-off-by: Arun Bharadwaj <arun@gumstix.com>
---
 arch/arm/boot/dts/am335x-pepper-43r.dts | 55 ++++++++++++++-------------------
 1 file changed, 23 insertions(+), 32 deletions(-)

diff --git a/arch/arm/boot/dts/am335x-pepper-43r.dts b/arch/arm/boot/dts/am335x-pepper-43r.dts
index 6b44f95..2f26a33 100644
--- a/arch/arm/boot/dts/am335x-pepper-43r.dts
+++ b/arch/arm/boot/dts/am335x-pepper-43r.dts
@@ -101,12 +101,6 @@
 		reg = <0x50>;
 	};
 
-	audio_codec: tlv320aic3106@1b {
-		compatible = "ti,tlv320aic3106";
-		reg = <0x1b>;
-		ai3x-micbias-vg = <0x2>;
-	};
-
 	ds_rtc: ds1340@68 {
 		compatible = "dallas,ds1340";
 		reg = <0x68>;
@@ -117,28 +111,28 @@
 &i2c1 {
 	status = "okay";
 	pinctrl-names = "default";
-	pinctrl-0 = <&i2c1_pins>;
+	pinctrl-0 = <&i2c1_pins &imu_pins>;
 	clock-frequency = <400000>;
 
-
-	// no device tree bindings or driver for this chip...fake it
-	accel: mpu9150@69 {
-		compatible = "invensense,mpu6050";
-		pinctrl-names = "default";
-		pinctrl-0 = <&accel_pins>;
-		reg = <0x69>;
-		interrupt-parent = <&gpio1>;
-		interrupt = <2 1>;
+	audio_codec: tlv320aic3106@1b {
+		compatible = "ti,tlv320aic3106";
+		reg = <0x1b>;
+		ai3x-micbias-vg = <0x2>;
 	};
 
-};
+	/* Accelerometer/Magnetometer */
+	lsm303d: lsm303d@1e {
+		compatible = "lsm303d";
+		int1-gpio = <&gpio2 4 GPIO_ACTIVE_HIGH>;
+		int2-gpio = <&gpio0 30 GPIO_ACTIVE_HIGH>;
+		reg = <0x1e>;
+	};
 
-/* Accelerometer */
-&am33xx_pinmux {
-	accel_pins: pinmux_accel {
-		pinctrl-single,pins = <
-			0x08 (PIN_INPUT | MUX_MODE7)   /* gpmc_ad2.gpio1_2 */
-		>;
+	/* Gyroscope */
+	l3gd20h: l3gd20h@6a {
+		compatible = "l3gd20h_gyr";
+		int-gpio = <&gpio1 2 GPIO_ACTIVE_HIGH>;
+		reg = <0x6a>;
 	};
 };
 
@@ -657,13 +651,13 @@
 	led@1 {
 		label = "pepper:user0:red";
 		gpios = <&gpio1 21 GPIO_ACTIVE_HIGH>;
-		linux,default-trigger = "none";
+		linux,default-trigger = "heartbeat";
 		default-state = "off";
 	};
 	led@0 {
 		label = "pepper:user0:blue";
 		gpios = <&gpio1 20 GPIO_ACTIVE_HIGH>;
-		linux,default-trigger = "none";
+		linux,default-trigger = "mmc0";
 		default-state = "off";
 	};
 };
@@ -706,14 +700,11 @@
 			0x16C (PIN_INPUT_PULLUP | MUX_MODE3)	/* uart0_rtsn.i2c1_scl */
 		>;
 	};
-	sd_pins: pinmux_sd_card {
+	imu_pins: pinmux_imu_pins {
 		pinctrl-single,pins = <
-			0xf0 (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_dat0.mmc0_dat0 */
-			0xf4 (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_dat1.mmc0_dat1 */
-			0xf8 (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_dat2.mmc0_dat2 */
-			0xfc (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_dat3.mmc0_dat3 */
-			0x100 (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_clk.mmc0_clk */
-			0x104 (PIN_INPUT_PULLUP | MUX_MODE0)	/* mmc0_cmd.mmc0_cmd */
+			0x8  (PIN_INPUT | MUX_MODE7)    /* gpmc_ad2.gpio1_2 */
+			0x98 (PIN_INPUT | MUX_MODE7)    /* gpmc_wen.gpio2_4 */
+			0x70 (PIN_INPUT | MUX_MODE7)    /* gpmc_wait0.gpio0_30 */
 		>;
 	};
 };
-- 
2.7.4

